data_model/src/events/pipeline.rs

```diff
@@ -126,11 +133,48 @@ pub enum Status {
     Committed,
 }
 
+/// Type of [`Status`].
+#[derive(
+    Debug,
+    PartialOrd,
+    Ord,
+    PartialEq,
+    Eq,
+    Clone,
+    Copy,
+    Decode,
+    Encode,
+    IntoSchema,
+    Hash,
+    Serialize,
+    Deserialize,
+)]
+pub enum StatusType {
+    /// Represents [`Status::Validating`].
+    Validating,
+    /// Represents [`Status::Rejected`].
+    Rejected,
+    /// Represents [`Status::Committed`].
+    Committed,
+}
+
+impl Status {
+    fn type_(&self) -> StatusType {
```

Comment:
```suggestion
    fn kind(&self) -> StatusKind {
```
While this is not related to Haskell's `kind` system, I'd prefer not having names which collide with `rust` keywords. Especially since the standard way of naming a function called `type` is 
```rust 
r#type
```